### PR TITLE
DAT-3002 Split the GlobalTitle between environments

### PR DESCRIPTION
**Pull request by sebastianmarzjan**

### CODE REVIEW

File: extensions/wikia/WikiFactory/WikiFactory.php [hunk 1]
```diff
@@ -1152,59 +1160,42 @@ static public function getLocalEnvURL( $url ) {
...
-		if ( preg_match( $regexp, $server, $groups ) === 1 ) {
-			// devbox
-			$devbox = $groups[1];
-			$server = str_replace( '.' . $devbox . '.wikia-dev.com', '', $server );
-		} else {
-			$server = str_replace( '.wikia.com', '', $server );
+		// strip env-specific pre- and suffixes for staging environment
+		$server = mb_ereg_replace( '^(preview|verify|sandbox-[a-z0-9]+)\.', '', $server );
+		if( $wgWikiaEnvironment === WIKIA_ENV_DEV ) {
+			$server = str_replace( $wgDevelEnvironmentName . '.wikia-dev.com', '', $server );
```
Comment (SebastianMarzjan) [inline extensions/wikia/WikiFactory/WikiFactory.php]: This needs to be handled differently to be able to transform URL from context of devboxA to devboxB

File: extensions/wikia/WikiFactory/WikiFactory.php [hunk 2]
```diff
@@ -1146,65 +1154,52 @@ static public function getLocalEnvURL( $url ) {
...
-		//  wikiname.developer.wikia-dev.com
-
-		$servers = [ 'preview.', 'sandbox-s1.', 'verify.' ];
-		foreach ( $servers as $serv ) {
-			if ( strpos( $server, $serv ) === 0 ) {
-				$server = substr( $server, strlen( $serv ) );
-			}
-		}
+		// strip env-specific pre- and suffixes for staging environment
+		$server = mb_ereg_replace( '^(preview|verify|sandbox-[a-z0-9]+)\.', '', $server );
```
Comment (macbre) [inline extensions/wikia/WikiFactory/WikiFactory.php]: Any reason why `mb_ereg_replace` is used here instead of `preg_replace`?

**Comment by macbre** [comment]
LGTM

File: extensions/wikia/WikiFactory/WikiFactory.php (hunk 2 above)
Comment (SebastianMarzjan) [inline extensions/wikia/WikiFactory/WikiFactory.php]: None :)

**Pull request merged**